### PR TITLE
MacIdeAppleEvents: add an apple event for getting current solution path

### PR DESCRIPTION
**Pull request by abock**
Unity needs to be able to query running instances of MD/VS Mac for the active/selected current solution path.

/cc @jbevain

### CODE REVIEW

**Comment by jbevain** [comment]
Previously Unity would open their C# project in the first VSM instance it could find.
I've verified that we can use this feature to detect the proper VSM instance to work with.
A back port to 8.5 would be appreciated as well.

File: main/src/addins/MacPlatform/query-vsmac.mm [hunk 1]
```diff
@@ -0,0 +1,204 @@
+//
```
Comment (abock) [inline main/src/addins/MacPlatform/query-vsmac.mm]: Note that this is just a small reference utility/library to test the feature. This does not actually build or ship as part of VS Mac itself.

**Comment by Therzok** [comment]
@jbevain we are still merging master to 8.5, so it will be picked up automatically.

**Comment by Therzok** [comment]
LGTM, merge at will.

**Comment by abock** [comment]
Would be good to get this for 8.4 though, which is what I think @jbevain was thinking.

**Comment by abock** [comment]
@monojenkins backport release-8.4

**Pull request merged**